### ISSUE TITLE
Bug in delta_r coffea.0.6.47

### ISSUE THREAD

**Issue by areinsvo** [scikit-hep/coffea#395]
**Describe the bug**
delta_r does not work if the second array is also nested. 

**To Reproduce**
```
events = NanoEventsFactory.from_file("root://cmseos.fnal.gov//store/user/cmsdas/2021/long_exercises/TTGamma/TestFiles/TTGamma_1l.root", schemaclass=NanoAODSchema).events()
events.Muon.delta_r(events.Photon)
```

**Output**
Error message:
```
ValueError                                Traceback (most recent call last)
<ipython-input-127-c220807612a8> in <module>
----> 1 events.Muon.delta_r(events.Photon)

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/coffea/nanoevents/methods/vector.py in delta_r(self, other)
    416         :math:`\sqrt{\Delta\eta^2 + \Delta\phi^2}`
    417         """
--> 418         return numpy.sqrt(self.delta_r2(other))
    419 
    420     def nearest(

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/coffea/nanoevents/methods/vector.py in delta_r2(self, other)
    409     def delta_r2(self, other):
    410         """Squared `delta_r`"""
--> 411         return (self.eta - other.eta) ** 2 + self.delta_phi(other) ** 2
    412 
    413     def delta_r(self, other):

/cvmfs/sft.cern.ch/lcg/views/LCG_96python3/x86_64-centos7-gcc8-opt/lib/python3.6/site-packages/numpy/lib/mixins.py in func(self, other)
     23         if _disables_array_ufunc(other):
     24             return NotImplemented
---> 25         return ufunc(self, other)
     26     func.__name__ = '__{}__'.format(name)
     27     return func

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/awkward1/highlevel.py in __array_ufunc__(self, ufunc, method, *inputs, **kwargs)
   1367         See also #__array_function__.
   1368         """
-> 1369         return awkward1._connect._numpy.array_ufunc(ufunc, method, inputs, kwargs)
   1370 
   1371     def __array_function__(self, func, types, args, kwargs):

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/awkward1/_connect/_numpy.py in array_ufunc(ufunc, method, inputs, kwargs)
    213 
    214     out = awkward1._util.broadcast_and_apply(
--> 215         inputs, getfunction, behavior, allow_records=False
    216     )
    217     assert isinstance(out, tuple) and len(out) == 1

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/awkward1/_util.py in broadcast_and_apply(inputs, getfunction, behavior, allow_records, left_broadcast, right_broadcast, numpy_to_regular)
   1008     else:
   1009         isscalar = []
-> 1010         out = apply(broadcast_pack(inputs, isscalar), 0)
   1011         assert isinstance(out, tuple)
   1012         return tuple(broadcast_unpack(x, isscalar) for x in out)

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/awkward1/_util.py in apply(inputs, depth)
    771                         nextinputs.append(x)
    772 
--> 773                 outcontent = apply(nextinputs, depth + 1)
    774                 assert isinstance(outcontent, tuple)
    775 

/uscms_data/d3/ahall/DAS2021/fork/TTGamma_LongExercise/ttgenv/lib/python3.6/site-packages/awkward1/_util.py in apply(inputs, depth)
    812                         nextinputs.append(fcn(x, offsets))
    813                     elif isinstance(x, listtypes):
--> 814                         nextinputs.append(x.broadcast_tooffsets64(offsets).content)
    815                     # handle implicit left-broadcasting (unlike NumPy)
    816                     elif left_broadcast and isinstance(x, awkward1.layout.Content):

ValueError: in ListOffsetArray64, cannot broadcast nested list

(https://github.com/scikit-hep/awkward-1.0/blob/0.4.5/src/cpu-kernels/awkward_ListArray_broadcast_tooffsets.cpp#L27)`
```
**Desktop (please complete the following information):**
 - OS: SL7 on the LPC
 - Version 0.6.47

**Additional context**
My temporary workaround is to define my own method using ak.cartesian:
```
        def dR(one,two):
            pairs = ak.cartesian({"i0":one,"i1":two},nested=True)
            diffPhi = abs(pairs.i0.phi - pairs.i1.phi)
            diffEta = abs(pairs.i0.eta - pairs.i1.eta)
            return np.sqrt(diffPhi*diffPhi + diffEta*diffEta)
```

**Comment by lgray** [comment]
@areinsvo The answer here is in two parts.

You probably want to use this for finding the nearest deltaR match? For that specific case there is now the `nearest` function.
https://github.com/CoffeaTeam/coffea/blob/master/coffea/nanoevents/methods/vector.py#L420
You would use it like:
```python3
events.Muon.nearest(events.Photon, threshold=0.4)
```
This will return to you a list of best matching candidates according to some metric. The default metric is delta_r, there is also an option to return the value.

In the more general case where you want all the delta_r combinatorics, I guess this warrants a new function in vector.py that does just that (essentially it's nearest without the minimization).

If you really just need this for the nearest-DR matching then I'll make this a wontfix rather than a bug, since delta_r in vector is supposed to operate on guaranteed pairs of four vectors (it literally just calculates deltar and nothing else). 

If you need it for more I'll reclassify as enhancement since what you're actually asking for is a function that makes output product tables of some metric.

**Comment by lgray** [comment]
I went ahead and made a new function called `metric_table` that does the more general operation, and it will be included with #386. 

If you need a backport of this to 0.6.x let me know.

**Comment by areinsvo** [comment]
To document: the `nearest` function does what I need it to do, in combination with `ak.fill_none` to deal with events with 0 photons. I'll go ahead and close this issue.
